include/odp/api/spec/event_vector.h

```diff
@@ -0,0 +1,249 @@
+ */
+uint32_t odp_event_vector_size(odp_event_vector_t evv);
+
+/**
+ * Type of events stored in event vector
+ *
+ * If all events in the vector are of the same type, function returns the
+ * particular event type. If the vector is empty or includes multiple event
+ * types, ODP_EVENT_ANY is returned instead.
+ *
+ * @param      evv        Event vector handle
+ *
+ * @return Event type
+ */
+odp_event_type_t odp_event_vector_type(odp_event_vector_t evv);
+
+/**
+ * Set the number of events stored in a vector
+ *
+ * Update the number of events stored in a vector. When the application is
+ * producing an event vector, this function shall be used by the application
+ * to set the number of events available in this vector.
+ *
+ * The maximum number of events this vector can hold is defined by
+ * odp_pool_param_t::event_vector.max_size. The size value must not be greater
+ * than odp_pool_param_t::event_vector.max_size
+ *
+ * All handles in the vector table (0 .. size - 1) need to be valid event
+ * handles.
+ *
+ * @param evv    Event vector handle
+ * @param size   Number of events in this vector
+ *
+ * @see odp_event_vector_tbl()
+ */
+void odp_event_vector_size_set(odp_event_vector_t evv, uint32_t size);
+
+/**
+ * Event vector user area
+ *
+ * Returns pointer to the user area associated with the event vector. Size of
+ * the area is fixed and defined in vector pool parameters.
+ *
+ * @param  evv  Event vector handle
+ *
+ * @return       Pointer to the user area of the event vector
+ * @retval NULL  The event vector does not have user area
+ */
+void *odp_event_vector_user_area(odp_event_vector_t evv);
+
+/**
+ * Check user flag
+ *
+ * Implementation clears user flag during new event vector creation (e.g. alloc
+ * and event input) and reset. User may set the flag with
+ * odp_event_vector_user_flag_set(). Implementation never sets the flag, only
+ * clears it. The flag may be useful e.g. to mark when the user area content is
+ * valid.
+ *
+ * @param evv  Event vector handle
+ *
+ * @retval 0    User flag is clear
+ * @retval >0   User flag is set
+ */
+int odp_event_vector_user_flag(odp_event_vector_t evv);
+
+/**
+ * Set user flag
+ *
+ * Set (or clear) the user flag.
+ *
+ * @param evv  Event vector handle
+ * @param val  New value for the flag. Zero clears the flag, other values set
+ *             the flag.
+ */
+void odp_event_vector_user_flag_set(odp_event_vector_t evv, int val);
+
+/**
+ * Check that event vector is valid
+ *
+ * This function can be used for debugging purposes to check if an event vector
+ * handle represents a valid event vector. The level of error checks depends on
+ * the implementation. Considerable number of CPU cycles may be consumed
+ * depending on the level. The call should not crash if the event vector handle
+ * is corrupted.
+ *
+ * @param evv Event vector handle
+ *
+ * @retval 0 Event vector is not valid
+ * @retval 1 Event vector is valid
+ */
+int odp_event_vector_valid(odp_event_vector_t evv);
```

Comment:
This may not be needed. Today "is valid" functions are defined only to event, buffer, packet and packet_vector. "to_u64" and debug print functions could be enough for debugging.

Reply:
removed from v6